_studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw_vaapi.cpp

```diff
@@ -329,7 +329,7 @@ mfxStatus VideoDECODEVP8_HW::PackHeaders(mfxBitstream *p_bistream)
 
      //////////////////////////////////////////////////////////////////
      UMC::UMCVACompBuffer* compBufBs;
-     uint8_t *bistreamData = (uint8_t *)m_p_video_accelerator->GetCompBuffer(VASliceDataBufferType, &compBufBs, p_bistream->DataLength - offset);
+     uint8_t *bistreamData = (uint8_t *)m_p_video_accelerator->GetCompBuffer(VASliceDataBufferType, &compBufBs, p_bistream->DataLength + offset);
```

Comment:
perhaps just "size - offset instaead of "p_bistream->DataLength + offset"  ? Becuase this length is already used at least in two places here:
```
     // number of bytes in the slice data buffer for the partitions
     sliceParams->slice_data_size = (int32_t)size - offset;

```
`compBufBs->SetDataSize((int32_t)size - offset);`

Reply:
Hi Dmitry, code updated.
For the latest change, created buffer can contain valid data.